mono/metadata/icall-table.c

```diff
@@ -0,0 +1,426 @@
+/**
+ * \file
+ *
+ * Authors:
+ *   Dietmar Maurer (dietmar@ximian.com)
+ *   Paolo Molaro (lupus@ximian.com)
+ *	 Patrik Torstensson (patrik.torstensson@labs2.com)
+ *   Marek Safar (marek.safar@gmail.com)
+ *   Aleksey Kliger (aleksey@xamarin.com)
+ *
+ * Copyright 2001-2003 Ximian, Inc (http://www.ximian.com)
+ * Copyright 2004-2009 Novell, Inc (http://www.novell.com)
+ * Copyright 2011-2015 Xamarin Inc (http://www.xamarin.com).
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+#include <config.h>
+#include <glib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <ctype.h>
+#ifdef HAVE_ALLOCA_H
+#include <alloca.h>
+#endif
+#ifdef HAVE_SYS_TIME_H
+#include <sys/time.h>
+#endif
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#if defined (HAVE_WCHAR_H)
+#include <wchar.h>
+#endif
+
+#include <mono/metadata/icall-table.h>
+#include <mono/utils/mono-publib.h>
+#include <mono/utils/bsearch.h>
+
+/*
+ * icall.c defines a lot of icalls as static, to avoid having to add prototypes for
+ * them, just don't include any mono headers and emit dummy prototypes.
+ */
+// Generate prototypes
+#define ICALL_TYPE(id,name,first)
+#define ICALL(id,name,func) extern void func (void);
+#define HANDLES(inner) inner
+#include "metadata/icall-def.h"
+
+// Generate Icall_ constants
+#undef ICALL_TYPE
+#undef ICALL
+#undef HANDLES
+#define ICALL_TYPE(id,name,first)
+#define ICALL(id,name,func) Icall_ ## id,
```

Comment:
I'd prefer spaces after commas, but again just style.

Reply:
Its code from the old version.